apps/api/src/core/providers/http-sdk.provider.ts

```diff
@@ -13,20 +13,33 @@ import {
 } from "@akashnetwork/http-sdk";
 import { container } from "tsyringe";
 
-import { apiNodeUrl, nodeApiBasePath } from "@src/utils/constants";
+import { ChainNetworkConfigService } from "@src/core/services/chain-network-config/chain-network-config.service";
+import { nodeApiBasePath } from "@src/utils/constants";
 
-const SERVICES = [
-  BalanceHttpService,
-  AuthzHttpService,
-  BlockHttpService,
-  BidHttpService,
-  DeploymentHttpService,
-  LeaseHttpService,
-  ProviderHttpService,
-  CosmosHttpService
-];
-
-SERVICES.forEach(Service => container.register(Service, { useValue: new Service({ baseURL: apiNodeUrl }) }));
+container.register(BalanceHttpService, {
+  useFactory: c => new BalanceHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(AuthzHttpService, {
+  useFactory: c => new AuthzHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(BlockHttpService, {
+  useFactory: c => new BlockHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(BidHttpService, {
+  useFactory: c => new BidHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(DeploymentHttpService, {
+  useFactory: c => new DeploymentHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(LeaseHttpService, {
+  useFactory: c => new LeaseHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(ProviderHttpService, {
+  useFactory: c => new ProviderHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
+container.register(CosmosHttpService, {
+  useFactory: c => new CosmosHttpService({ baseURL: c.resolve(ChainNetworkConfigService).getBaseAPIUrl() })
+});
```

Comment:
i don't think i understand this change. this is a repetitive code, could still iterate over every service in array